### DISCUSSION TITLE
kfraft mode and metrics

### DISCUSSION THREAD

**Discussion by msillence** [strimzi/strimzi-kafka-operator#9842] [General]
Hi, 

I noticed the warning:

The .spec.zookeeper section in the Kafka custom resource is ignored
        in KRaft mode and should be removed from the custom resource.

Looking at what was in there I see metricsConfig which I don't see supported by the [NodePoolSpec](https://strimzi.io/docs/operators/latest/configuring#type-KafkaNodePoolSpec-reference)

Should there be something for that?

Thanks,
Martin

**Comment by scholzj** [comment]
I'm not sure I follow what exactly you mean. The only metrics configuration in `.spec.zookeeper` are the ZooKeepr metrics. And you do not use / need those when running in KRaft.

**Reply by msillence** [reply]
I mean the promethius exporter part https://strimzi.io/docs/operators/latest/configuring#con-common-configuration-prometheus-reference
I presume there will still be metrics we can collect from the controller running in kraft mode

**Reply by scholzj** [reply]
Sorry, I still don't follow. The metrics configuration under `.spec.zookeeper` is only for ZooKeeper. The one for Kafka is in `.spec.kafka` Nothing changed with that one it works as before.

**Reply by msillence** [reply]
Great, thank you so the controller and broker both get their metrics config from the kafka spec.

**Reply by scholzj** [reply]
Yes, correct. They share the same configuration for the JMX Exporter.

**Reply by scholzj** [reply]
PS: If you think they should be configured per node pool, please feel free to explain why (either here or open an enhancement issue) and we can look into it. I do not think there was any detailed discussion or consideration about that so far.

**Reply by msillence** [reply]
No, it's fine I was just confused how it was meant to be configured after seeing the warning to remove the zookeeper section of the config.
Of note I did try removing the zookeeper section and it failed to deploy saying it was required so I put it back in.
Thanks again for the clarification

**Reply by scholzj** [reply]
It is not required from 0.40.0. So you should make sure that is the Strimzi version you are running and that you updated the CRDs as well.
